blockchain/validate.go

```diff
@@ -752,14 +752,35 @@ func (b *BlockChain) checkBlockContext(block *btcutil.Block, prevNode *blockNode
 
 	fastAdd := flags&BFFastAdd == BFFastAdd
 	if !fastAdd {
+		// Obtain the latest state of the deployed CSV soft-fork in
+		// order to properly guard the new validation behavior based on
+		// the current BIP 9 version bits state.
+		csvState, err := b.deploymentState(prevNode, chaincfg.DeploymentCSV)
+		if err != nil {
+			return err
+		}
+
+		// Once the CSV soft-fork is fully active, we'll switch to
+		// using the current median time past of the past block's
+		// timestamps for all lock-time based checks.
+		blockTime := header.Timestamp
+		if csvState == ThresholdActive {
+			medianTime, err := b.calcPastMedianTime(prevNode)
```

Comment:
I believe this is incorrect.  The `calcPastMedianTime` takes the block node you want to the calulate the median time for as opposed to its parent.  However, I don't think it's available in this function at the current time.  It'll probably need to be modified to pass in the current block node and get the previous one for the places that need it such as `checkBlockHeaderContext` and `deploymentState`.

This also likely means the block node creation for the block being validated will need to move up before the call to `checkBlockContext` in `accept.go`.